src/TestParticle.jl

```diff
@@ -62,9 +62,43 @@ function getinterp(A, gridx, gridy, gridz)
       return SA[interpx(r...), interpy(r...), interpz(r...)]
    end
 
-   return get_field
+   return Field(get_field)
 end
 
+function is_time_dependent(f)
+   itp_param_number = 2
+   numargs = [length(m.sig.parameters)-1 for m in methods(f)]
+   itp = any(x -> x == itp_param_number, numargs)
+   if !itp
+      if any(x -> x == itp_param_number-1, numargs)
+         return false
+      else
+         name = nameof(f)
+         throw(ArgumentError("All methods for the field function $name had too many arguments."))
+      end
+   else
+      return true
+   end
+end
+
+abstract type AbstractField{itd} end
+
+struct Field{itd, F} <: AbstractField{itd}
+   field_function::F
+   Field{itd, F}(field_function::F) where {itd, F} = isa(itd, Bool) ? new(field_function) : throw(ArgumentError("itd must be a boolean."))
+end
+
+Field(f::Function) = Field{is_time_dependent(f), typeof(f)}(f)
+
+(f::AbstractField{true})(xu, t) = f.field_function(xu, t)
+(f::AbstractField{true})(xu) = throw(ArgumentError("Time-dependent field function must have a time argument."))
+(f::AbstractField{false})(xu, t) = f.field_function(xu)
+(f::AbstractField{false})(xu) = f.field_function(xu)
```

Comment:
I guess we don't need this case, as `t` is always included when passing to the ODE solvers?

Reply:
Indeed, the package doesn't need this method. But I still want to preserve this method. There are two reasons:
1. User defines a method accepting an argument for a time-independent field, but the method disappears after the wrapping process. I think it is counter-intuitive. What if they want to unpack those parameters and implement one themselves?
2. It will make those methods more symmetricial.

The test for this line has been appended to this PR.